pkg/sources/forward/data_forward.go

```diff
@@ -207,6 +216,32 @@ func (isdf *DataForward) forwardAChunk(ctx context.Context) {
 	// Process only if we have any read messages.
 	// There is a natural looping here if there is an internal error while reading, and we are not able to proceed.
 	if len(readMessages) == 0 {
+		// calculate that watermark is idling or not for the given maxWait duration
+		// If no any message is coming and watermark is idling then start increasing the watermark with "minIncrement" config
+		// value after waiting for each "maxWait" duration.
+		if isdf.isWatermarkIdle() {
+			// wait for maxWait duration before publishing the idle watermark with MinIncrement value.
+			if time.Since(isdf.idleWMFound) < isdf.WatermarkConfig.IdleSource.GetMaxDelay() {
+				return
+			}
+			// reset the idleWMFound time while publishing the idle watermark.
+			isdf.idleWMFound = time.Time{}
+			// MinIncrement is the value to be added in idle watermark while publishing.
+			minIncrement := isdf.WatermarkConfig.IdleSource.GetMinIncrement()
+			// Get the head watermark value then add the minIncrement to it and publish the idle watermark with updated value.
+			headWatermark := isdf.wmFetcher.ComputeHeadWatermark(0)
+			updatedWM := headWatermark.Add(*minIncrement)
+			// check if updated watermark is greater than the current time then reset the updated watermark with current time.
+			if updatedWM.After(time.Now()) {
+				updatedWM = time.Now()
+			}
+			isdf.srcWMPublisher.PublishIdleWatermarks(updatedWM)
+			// set the time for last updated watermark which will be used for calculating maximum wait time is passed?
+			isdf.lastUpdateWM = time.Now()
+			// reset the currentWatermark to nil which is used to calculate when watermark is idling.
+			isdf.currentWatermark = nil
```

Comment:
I can't fetch the WM value and set it here because it's containing the older valuer after publishing the new WM also, In the next iteration only I'm able to fetch updated value of WM

I'm not sure what's the reason behind it but that's what I have encountered!!!